doc/principles.md

```diff
@@ -0,0 +1,62 @@
+# Principles
+
+## Introduction
+
+Based on [KPConv-PyTorch](https://github.com/HuguesTHOMAS/KPConv-PyTorch), this library offers a deep learning model for points clouds classification. It is based on a Kernel Points Convolution (KPConv) algorithm, inspired by image-based convolution. The entry is made of a set of 3D-points, associated to a *feature* or *attribute*, which is here the nature of the point, in a pre-defined number of categories classification. 
+
+The objective of the model is to be able to classify a new 3D-points set with these categories of materials or objects.
+
+Each kernel point is associated to a weight function (called *kernel function*), applied to all points located in a radius around it. It is based on a correlation function quantifying the influence of the kernel point on the base of its distance to each point.
+
+The number of kernel points is not constrained and nor are their positions (in particular, they are not on a grid), which makes the design adaptable to the shape of the points cloud. That is why it is called a *deformable* KPConv algorithm. The position of kernel points is determined by an optimization problem, where each point applies a repulsive force on others. The process is calibrated so that there is a small overlap between each kernel area of influence and a good space coverage.
+
+As an intermediary result, there is a neighborhood matrix, containing for each point of the 3D-set (*N'* line) the reference to all points located in the neighborhood (*nmax* columns). Some colums can be empty, when the neighborhood contains less points.
+
+This library has been developed by Hugues Thomas and improved by Oslandia, in the particular context of building information modeling, for indoor and outdoor point segmentation (IASBIM project).
+
+## Indoor and outdoor available annotated data
+
+This algorithm has been used in a particular context of segmentation of indoor and outdoor data. 4 datasets have been used to test the neural network model:
```

Comment:
No, it was just a means to say that the original algorithm was calibrated on the 4 datasets, but since the CLI is only dedicated to S3DIS, I have shorten the description of others (see above).